swig/cpp/src/Struct.h

```diff
@@ -70,19 +70,19 @@ class Val
     Val(sr_val_t *val, S_Counter counter);
     Val(const char *val, sr_type_t type = SR_STRING_T);
     Val(bool bool_val, sr_type_t type = SR_BOOL_T);
-    Val(double decimal64_val, sr_type_t type = SR_DECIMAL64_T);
```

Comment:
I clearly lack SWIG experience here. I was bothered a small bit by the irregularity in these constructors. If I was designing a C++-only API, I would go with abandoning the type info tag and rely on native overload resolution rules, something like this:

```c++
explicit Val(const int8_t v);
explicit Val(const int16_t v);
explicit Val(const int32_t v);
explicit Val(const uint8_t v);
explicit Val(const uint16_t v);
explicit Val(const uint32_t v);
explicit Val(const double v);
//...
```
However, I have no clue if that would play with SWIG at all, so let's ignore this nitpicking for now.